recipes/libdrm/all/conanfile.py

```diff
@@ -0,0 +1,159 @@
+from conans import ConanFile, Meson, tools
+import os
+import shutil
+
+
+class LibdrmConan(ConanFile):
+    name = "libdrm"
+    description = "User space library for accessing the Direct Rendering Manager, on operating systems that support the ioctl interface"
+    topics = ("libdrm", "graphics")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://gitlab.freedesktop.org/mesa/drm"
+    license = "MIT"
+
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "libkms": [True, False],
+        "intel": [True, False],
+        "radeon": [True, False],
+        "amdgpu": [True, False],
+        "nouveau": [True, False],
+        "vmwgfx": [True, False],
+        "omap": [True, False],
+        "exynos": [True, False],
+        "freedreno": [True, False],
+        "tegra": [True, False],
+        "vc4": [True, False],
+        "etnaviv": [True, False],
+        "valgrind": [True, False],
+        "freedreno-kgsl": [True, False],
+        "udev": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "libkms": True,
+        "intel": True,
+        "radeon": True,
+        "amdgpu": True,
+        "nouveau": True,
+        "vmwgfx": True,
+        "omap": False,
+        "exynos": False,
+        "freedreno": True,
+        "tegra": False,
+        "vc4": True,
+        "etnaviv": False,
+        "valgrind": False,
+        "freedreno-kgsl": False,
+        "udev": False
+    }
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def build_requirements(self):
+        self.build_requires("meson/0.59.0")
+
+    def config_options(self):
+        if self.settings.os == 'Windows':
+            del self.options.fPIC
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+        if self.options.shared:
+            del self.options.fPIC
+
+    def requirements(self):
+        if self.options.intel:
+            self.requires("xorg/system")
```

Comment:
When disabling intel, will it work on Windows?
I would be surprised by that.

Reply:
It does not. I need to raise an exception

Reply:
Another question, does this recipe need `kernel-headers-generic`?

Reply:
I have no idea!

Reply:
How can I find this information?

Reply:
Analyzing the rpm scripts of distributions, or browsing the sources.
I found this in the headers:
https://gitlab.freedesktop.org/mesa/drm/-/blob/main/include/drm/drm.h#L38-L41
I think those are provided by the kernel headers.